src/main/java/org/openrewrite/java/spring/boot2/LoggingShutdownHooksRecipe.java

```diff
@@ -0,0 +1,74 @@
+package org.openrewrite.java.spring.boot2;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.maven.MavenVisitor;
+import org.openrewrite.maven.tree.Maven;
+import org.openrewrite.maven.tree.Pom;
+import org.openrewrite.maven.tree.Scope;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.Collection;
+
+/**
+ * @author Alex Boyko
+ */
+public class LoggingShutdownHooksRecipe extends Recipe {
+
+    private static final String NEEDS_SEARCH_MARKER = "NEEDS_SEARCH_MARKER";
+
+    private static final XPathMatcher PROJECT_MATCHER = new XPathMatcher("/project");
+
+    private static final String SEARCH_MARKER_DESCRIPTION = "LoggingShutDownHook";
+
+    @Override
+    public String getDisplayName() {
+        return "Logging Shutdown Hooks";
+    }
+
+    @Override
+    public String getDescription() {
+        return "We now register a logging shutdown hook by default for jar based applications to ensure that " +
+                "logging resources are released when the JVM exits. If your application is deployed as a war then " +
+                "the shutdown hook is not registered since the servlet container usually handles logging concerns" +
+                "\n\n" +
+                "Most applications will want the shutdown hook. However, if your application has complex context" +
+                " hierarchies, then you may need to disable it. You can use the logging.register-shutdown-hook" +
+                " property to do that.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MavenVisitor() {
+            @Override
+            public Maven visitMaven(Maven maven, ExecutionContext ctx) {
+                Collection<Pom.Dependency> deps = maven.getModel().getDependencies(Scope.Compile);
+                if (deps.stream()
+                        .filter(d -> d.getArtifactId().equals("spring-boot")
+                                && d.getVersion().startsWith("2.5.")
+                                && d.getGroupId().equals("org.springframework.boot"))
+                        .findFirst()
+                        .isPresent()
+                ) {
```

Comment:
for this recipe the maven dependency check can be the [applicable test](https://github.com/openrewrite/rewrite/blob/6809be2616697debf5360337dc656f29efa57503/rewrite-core/src/main/java/org/openrewrite/Recipe.java#L169-L180)

> A recipe can optionally include an applicability test that can be used to determine whether it should run on a
   set of source files (or even be listed in a suggested list of recipes for a particular codebase).

Reply:
thanks for catching my error.  the `singleSourceApplicability` test is where it belongs, then tags are only visited when the maven pom has a boot 2.5 dependency